From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jon Hunter <jonathanh@nvidia.com>
Date: Mon, 14 Oct 2024 12:34:18 +0100
Subject: [PATCH 044/240] NVIDIA: SAUCE: net: phy: aquantia: Fix PHY interfaces

In Linux v6.1, commit 7de26bf144f6 ("net: phy: aquantia: Add some
additional phy interfaces") added support for additional PHY interfaces
and downstream commit f87ded88a11c ("NVIDIA: SAUCE: net: phy: aqr113c:
Enable Wake on boot configuration.") incorrectly removes some of these
from the function aqr107_config_init() when applied on top of v6.x
kernels. Fix this by reinstating these PHY interfaces.

Bug 3270211

Change-Id: I65fb191cccafea662350c2cfce1b6cf42a1ce756
Signed-off-by: Jon Hunter <jonathanh@nvidia.com>
Reviewed-on: https://git-master.nvidia.com/r/c/3rdparty/canonical/linux-noble/+/3229170
GVS: buildbot_gerritrpt <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Revanth Kumar Uppala <ruppala@nvidia.com>
(cherry picked from commit 65b5dbebef09b42543c3e74da4f63a18fc2b3e8a)

diff --git a/drivers/net/phy/aquantia/aquantia_main.c b/drivers/net/phy/aquantia/aquantia_main.c
index e956d29b6493b..2ae16db806da5 100644
--- a/drivers/net/phy/aquantia/aquantia_main.c
+++ b/drivers/net/phy/aquantia/aquantia_main.c
@@ -918,11 +918,15 @@ static int aqr_gen1_config_init(struct phy_device *phydev)
 
 	/* Check that the PHY interface type is compatible */
 	if (phydev->interface != PHY_INTERFACE_MODE_SGMII &&
+	    phydev->interface != PHY_INTERFACE_MODE_1000BASEKX &&
 	    phydev->interface != PHY_INTERFACE_MODE_2500BASEX &&
 	    phydev->interface != PHY_INTERFACE_MODE_XGMII &&
 	    phydev->interface != PHY_INTERFACE_MODE_USXGMII &&
+	    phydev->interface != PHY_INTERFACE_MODE_10G_QXGMII &&
 	    phydev->interface != PHY_INTERFACE_MODE_10GKR &&
-	    phydev->interface != PHY_INTERFACE_MODE_10GBASER)
+	    phydev->interface != PHY_INTERFACE_MODE_10GBASER &&
+	    phydev->interface != PHY_INTERFACE_MODE_XAUI &&
+	    phydev->interface != PHY_INTERFACE_MODE_RXAUI)
 		return -ENODEV;
 
 	WARN(phydev->interface == PHY_INTERFACE_MODE_XGMII,
-- 
2.43.0

